evm/utils/test_builder/builder_utils.py

```diff
@@ -0,0 +1,126 @@
+import copy
+import itertools
+import random
+import subprocess
+
+from collections.abc import (
+    Mapping,
+)
+
+from evm.db.backends.memory import MemoryDB
+
+from cytoolz import (
+    first,
+    merge_with,
+)
+from eth_utils import (
+    force_text,
+    int_to_big_endian,
+)
+from evm.utils.padding import (
+    pad32,
+)
+
+from eth_keys import keys
+
+try:
+    from vyper.compile_lll import (
+        compile_to_assembly,
+        assembly_to_evm,
+    )
+    from vyper.parser.parser_utils import LLLnode
+except ImportError:
+    vyper_available = False
+else:
+    vyper_available = True
+
+
+random.seed(0)
+
+
+def merge_if_dicts(values):
+    if all(isinstance(item, Mapping) for item in values):
+        return merge_with(merge_if_dicts, *values)
+    else:
+        return values[-1]
+
+
+def deep_merge(*dicts):
+    return merge_with(merge_if_dicts, *dicts)
+
+
+def is_cleanly_mergable(*dicts):
+    if len(dicts) <= 1:
```

Comment:
This function, while not a public API might benefit from a docstring since it's not entirely clear what *cleanly mergable* means without reading the implementation.  A human readable description of this function would go a long way to helping future readers understand it's intended functionality.